cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -288,7 +290,12 @@ public void symlink(Path source, Path targetLink) {
         this.context.debug("Deleting symbolic link to be re-created at {}", targetLink);
         Files.delete(targetLink);
       }
-      Files.createSymbolicLink(targetLink, source);
+      if (this.context.getSystemInfo().getOs().equals(WINDOWS)) {
+        Runtime rt = Runtime.getRuntime();
+        rt.exec("cmd //c 'mklink /D /J $(cygpath -w " + source + ") $(cygpath -w" + targetLink + ")");
```

Comment:
IMHO it is better to do that in a dedicated catch block:
```
} catch (FileSystemException e) {
  info("Failed to create native symlink due to lack of permissions. See https://github.com/devonfw/IDEasy/blob/main/documentation/symlinks.asciidoc for further details. Error was {}", e.getMessage());
  // workaround goes here...
}
```

For the workaround we are using `ProcessBuilder` that we typically wrap via `ProcessContext`.
Currently `FileAccessImpl` has no direct access to `IdeContext` and therefore `ProcessContext` (as I initially wanted to avoid cyclic dependencies as architectural best-practice), but we could easily change this.
Howerver, it is fully fine to use `ProcessBuilder` directly here. But just like `File` also `Runtime` should be considered deprecated as it has some bugs addressed by the new construct (`ProcessBuilder`).

Reply:
Well after rethinking this: Will the output of the `mklink` command go to nirvana (`/dev/null`) and potential errors (exit code other than 0) will be ignored then?
All this is already handled by `ProcessContext` so it may be the best option to use that.

Reply:
As now everything is solved, I assume this question is already cleared. However, the executable is just a `Path` but can be any command so also just `cmd` or `mklink`.